src/ui/views/Approval/components/PreCheckCard.tsx

```diff
@@ -0,0 +1,119 @@
+import { Skeleton } from 'antd';
+import { ExplainTxResponse } from 'background/service/openapi';
+import React from 'react';
+import IconCheck from 'ui/assets/icon-check.svg';
+import IconLoading from 'ui/assets/icon-loading.svg';
+import IconWarning, {
+  ReactComponent as IconRcWaring,
+} from 'ui/assets/icon-warning.svg';
+
+interface PreCheckCardProps {
+  loading: boolean;
+  version: ExplainTxResponse['pre_exec_version'];
+  data: ExplainTxResponse['pre_exec'];
+  errors?: {
+    msg: string;
+    code: number;
+  }[];
+  isReady: boolean;
+}
+
+const PreCheckCard = (props: PreCheckCardProps) => {
+  const { loading, version, data, errors, isReady } = props;
+  if (!isReady) {
+    return (
+      <div className="pre-check-card">
+        <div className="pre-check-card-header items-center mb-0 gap-[12px]">
+          <Skeleton.Avatar active style={{ width: 20, height: 20 }} />
+          <Skeleton.Input active style={{ width: 84, height: 15 }} />
+        </div>
+      </div>
+    );
+  }
+  if (loading) {
+    return (
+      <div className="pre-check-card">
+        <div className="pre-check-card-header items-center mb-0">
+          <img
+            src={IconLoading}
+            className="pre-check-card-icon is-loading"
+            alt=""
+          />
+          <div className="pre-check-card-text-loading">
+            Checking for errors...
+          </div>
+        </div>
+      </div>
+    );
+  }
+  if (['v0', 'v1'].includes(version)) {
+    return (
```

Comment:
This will make v0 and v1 version preExec always be success, even user native token is not enough to pay gas(i.e. `errors.length > 0`)